### PR TITLE
[Merged by Bors] - chore: add dependabot

### PR DESCRIPTION
**Pull request by pitmonticone**
This PR introduces a Dependabot configuration specifically for managing dependencies in GitHub Actions workflows.

It automates the process of checking for updates to GitHub Actions dependencies.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 2  # Specifies the version of the Dependabot configuration file format
+
+updates:
+  # Configuration for dependency updates
+  - package-ecosystem: "github-actions"  # Specifies the ecosystem to check for updates
+    directory: "/"  # Specifies the directory to check for dependencies; "/" means the root directory
+    schedule:
+      # Check for updates to GitHub Actions every week
+      interval: "weekly"
```
Comment (grunweg) [inline .github/dependabot.yml:9]: IIRC, other repositories like the queueboard has "month" here. Do you have a particular reason for the different frequency?

Comment (pitmonticone) [reply .github/dependabot.yml:9]: Nope, I can change that immediately.

Comment (pitmonticone) [reply .github/dependabot.yml]: Done @grunweg. Thanks.

**Review by grunweg — COMMENTED** [review]
This is my first time reviewing such a file --- cross-checking [github's documentation](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates) about [this](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#scheduleinterval), this looks sensible. Thanks for automating this!

~~Can you add some brief explanation what this does to the PR description? Thanks!~~ Never mind, you already did that.
maintainer merge

**Comment by bryangingechen** [comment]
Thanks!
bors r+

**Comment by bryangingechen** [comment]
Ah, it's not the end of the world, but since depandabot isn't aware of our `bors.in.yml` system, we won't be able to use its PRs as is for those updates. cf. #18017 .

I'll go through and do a manual update including all the updates now.

**Pull request closed without merging**